pattern/makepattern.py

```diff
@@ -45,9 +46,9 @@ def main(filenames):
 						if arm64.get_Rn(instr) in adrp_regs:
 							mask[i] &= ~(0xFFF << 10)
 
-			mask_str = ''.join(struct.pack('<I', m) for m in mask)
-			bits = ''.join(struct.pack('<I', i&m) for i,m in zip(instrs, mask))
-			regex = ''.join((re.escape(i) if m == '\xFF' else r'[\0-\xff]') for i, m in zip(bits, mask_str))
+			mask_str = b''.join(struct.pack('<I', m) for m in mask)
+			bits = b''.join(struct.pack('<I', i&m) for i,m in zip(instrs, mask))
+			regex = b''.join((re.escape(i.to_bytes(1)) if m == 0xff else b'[\0-\xff]') for i, m in zip(bits, mask_str))
```

Comment:
I wasn't able to find a way that works for both py2 and 3 without issues, so let's just use the old version for py2 and the new version for py3:

```suggestion
			if sys.version_info[0] > 2:
			    mask_str = b''.join(struct.pack('<I', m) for m in mask)
				bits = b''.join(struct.pack('<I', i & m) for i, m in zip(instrs, mask))
				regex = b''.join((re.escape(i.to_bytes(1)) if m == 0xff else b'[\0-\xff]') for i, m in zip(bits, mask_str))
			else:
				mask_str = ''.join(struct.pack('<I', m) for m in mask)
				bits = ''.join(struct.pack('<I', i & m) for i, m in zip(instrs, mask))
				regex = ''.join((re.escape(i) if m == '\xFF' else r'[\0-\xff]') for i, m in zip(bits, mask_str))
```